akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+  to improve performance and maintainability of your system. Do not depend on Java serialization for production deployments.
+
+The below snippet explains in the minimal amount of lines how a custom serializer can be registered.
+For more in-depth explanations on how serialization picks the serializer to use etc, please refer to its documentation.
+
+First we start by defining our domain model class, here representing a person:
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#simplest-custom-serializer-model
+
+Next we implement a serializer (or extend an existing one to be able to handle the new ``Person`` class):
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#simplest-custom-serializer
+
+And finally we register the serializer and bind it to handle the ``docs.persistence.Person`` class:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#simplest-custom-serializer-config
+
+Deserialization will be performed by the same serializer which serialized the message initially
+because of the ``identifier`` being stored together with the message.
+
+For more advanced techniques for using serializers refer to the :ref:`Akka Serialization <serialization-scala>` documentation,
+giving the :ref:`string-manifest-serializer-scala` special attention, as it is tremendously useful for Persistence based applications
+(as we will see in some of the below explained evolutions).
+
+
+Serializing to human-readable documents
+---------------------------------------
+Some projects may prefer to store events in human-readable formats instead of formats optimised for write/read throughput.
+The style of serialization used in Akka Persistence is completely left up to the concrete Journal plugin implementations -
+this means that some community provided journals may offer human readable serialization, however it is not the default style
+to persist events.
+
+Additionally some datastores are able to store "documents" as JSON (or BSON) natively, wich may have additional benefits
+for later debugging or searchability through the events. In general we highly encourage the use projections using
+:ref:`akka-persistence-query-scala` to migrate data stored using a binary format (optimised for write throughput)
+stored on the "write side" of the application, to the "read side" of the application – which more often than not
+is a completely different database, which for example is more suited for storing the events in a human readable format.
+
+One of the possible ways of supporting this is to emit "the type that the journal understands how to store natively"
+(e.g. a JSON type), in the ``EventAdapter`` such that the Journal can directly store it in the datastore without applying
+any additional serialization. This technique is described as sub-section of the :ref:`detach-domain-from-data-model-scala` pattern.
```

Comment:
...is described **in more detail as part of the** :ref:`detach-domain-from-data-model-scala` pattern.